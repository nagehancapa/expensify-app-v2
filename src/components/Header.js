import React from "react";
import { NavLink } from "react-router-dom";
import { useDispatch } from "react-redux";
import { startLogout } from "../store/user/actions";

const Header = () => {
  const dispatch = useDispatch();
  function onClick() {
    dispatch(startLogout());
  }

  return (
    <header>
      <h1>Expensify</h1>
      <NavLink end to="/dashboard">
        Dashboard
      </NavLink>
      <NavLink end to="/create">
        Create Expense
      </NavLink>
      <button onClick={onClick}>Logout</button>
    </header>
  );
};

export default Header;
