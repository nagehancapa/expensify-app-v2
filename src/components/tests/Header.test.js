import React from "react";
import * as reactRedux from "react-redux";
import { shallow } from "enzyme";
import Header from "../Header";

// Mock useDispatch hook
const useDispatch = jest.spyOn(reactRedux, "useDispatch");

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: jest.fn(),
}));

beforeEach(() => {
  useDispatch.mockClear();
});

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

test("should render Header correctly", () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();

  // expect(wrapper.find("h1").text()).toBe("Expensify");
  // const renderer = new ReactShallowRenderer();
  // renderer.render(<Header />);
  // expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test("should call onClick", () => {
  const dispatch = jest.fn();
  useDispatch.mockReturnValue(dispatch);
  const wrapper = shallow(<Header />);
  wrapper.find("button").simulate("click");
  expect(dispatch).toHaveBeenCalled();
});
