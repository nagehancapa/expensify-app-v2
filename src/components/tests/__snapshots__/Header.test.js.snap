// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    end={true}
    to="/dashboard"
  >
    Dashboard
  </NavLink>
  <NavLink
    end={true}
    to="/create"
  >
    Create Expense
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
